packages/expo-cli/src/commands/build-native/index.ts

```diff
@@ -70,12 +78,57 @@ export async function buildAction(projectDir: string, options: Options): Promise
   }
 }
 
-async function statusAction(projectDir: string): Promise<void> {
-  throw new Error('not implemented yet');
+async function statusAction(
+  projectDir: string,
+  { platform, status, buildId }: StatusOptions
+): Promise<void> {
+  const user: User = await UserManager.ensureLoggedInAsync();
+  const { exp }: ProjectConfig = getConfig(projectDir);
+
+  const accountName = exp.owner || user.username;
+  const projectName = exp.slug;
+
+  const projectId = await ensureProjectExistsAsync(user, {
+    accountName,
+    projectName,
+  });
+
+  const client = ApiV2.clientForUser(user);
+
+  const spinner = ora().start('Fetching build history...');
+
+  let builds: BuildInfo[] | undefined;
+
+  try {
+    const params = {
+      ...(['android', 'ios'].includes(platform) ? { platform } : null),
+      ...(status ? { status } : null),
+    };
+
+    if (buildId) {
+      const buildStatus = await client.getAsync(`projects/${projectId}/builds/${buildId}`, params);
```

Comment:
this endpoint does not take those params
pair (status + platform) and build options should be mutually exclusive
verify that before sending request